### DISCUSSION TITLE
Is there a way to generate the code outside of Bazel's output directory?

### DISCUSSION THREAD

**Discussion by c4milo** [rules-proto-grpc/rules_proto_grpc#120] [Q&A] [answered]
If so, can the option be given globally so we don't have to define it for every target? 

These rules are awesome, thank you!

**Comment by aaliddell** [comment, marked as answer]
Writing outside of the output root is generally counter to Bazel's architecture, especially since rules may be run more than once if there are multiple configurations.

e.g. if your build has two configurations (even just debug vs release), these two must write to separate files, otherwise you'll either end up with indetermined state in the file or Bazel will refuse to continue due to the same output path being produced by two different actions.

What would be the use case for needing to write to an alternative location?

**Reply by c4milo** [reply]
> What would be the use case for needing to write to an alternative location?

IDE support, mostly. Also to ease the transition to Bazel by still supporting people’s existing build tooling.

**Comment by c4milo** [comment]
FWIW, this may be common desired since some people like to review generated code too, and as I mentioned, have their IDEs working correctly. Here is another example: https://github.com/cockroachdb/cockroach/issues/58018
